src/components/views/dialogs/ChatInviteDialog.js

```diff
@@ -435,6 +445,39 @@ module.exports = React.createClass({
         return inviteList;
     },
 
+    _lookupThreepid(medium, address) {
+        // wait a bit to let the user finish typing
+        return q.delay(500).then(() => {
+            // If the query has changed, forget it
+            if (this.state.queryList[0] && this.state.queryList[0].address !== address) {
```

Comment:
I sort-of wonder if it would be better to set a flag when we are changing the addresslist, rather than do this the whole time: basically rolling your own promise cancellation.

something like:

```
_lookupThreepid: function (medium, address) {
    let cancelled = false;
    this.cancelThreepidLookup = function() {
        cancelled = true;
    }
    return q.delay(500).then(() => {
        if (cancelled) { return null }
        ...
    }) /* ... */ .finally(() => { this.cancelThreepidLookup = null })
}
```

then call `cancelThreepidLookup` in `onQueryChanged`